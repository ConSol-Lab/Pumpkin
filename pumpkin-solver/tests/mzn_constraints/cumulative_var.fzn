var 1..5: s1:: output_var;
var 3..5: s2:: output_var;
var 1..5: s3:: output_var;

var 1..2: p1:: output_var;
var 2..2: p2:: output_var;
var 1..2: p3:: output_var;

var 3..4: r1:: output_var;
var 2..2: r2:: output_var;
var 4..5: r3:: output_var;

var 1..4: c:: output_var;

constraint pumpkin_cumulative_vars([s1, s2, s3], [p1, p2, p3], [r1, r2, r3], c);

solve satisfy;
