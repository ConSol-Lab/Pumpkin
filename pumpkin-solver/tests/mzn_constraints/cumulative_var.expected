s1 = 2;
s2 = 3;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 3;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 4;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 4;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 4;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 4;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 4;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 4;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 4;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 4;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 4;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 4;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 4;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 4;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 4;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 4;
p1 = 1;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 2;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 2;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 3;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 3;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 4;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 4;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 4;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 4;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 4;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 4;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 2;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 2;
p1 = 2;
p2 = 2;
p3 = 1;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 3;
s3 = 5;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 4;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 4;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 4;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 1;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 4;
s2 = 5;
s3 = 2;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 2;
s2 = 5;
s3 = 3;
p1 = 1;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 5;
s2 = 3;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 3;
s3 = 5;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 3;
s2 = 5;
s3 = 1;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 3;
r2 = 2;
r3 = 4;
c = 4;
----------
s1 = 1;
s2 = 5;
s3 = 3;
p1 = 2;
p2 = 2;
p3 = 2;
r1 = 4;
r2 = 2;
r3 = 4;
c = 4;
----------
==========
